src/Microsoft.HttpRepl/Preferences/Preferences.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.HttpRepl.FileSystem;
+using Microsoft.HttpRepl.UserProfile;
+
+namespace Microsoft.HttpRepl.Preferences
+{
+    public class Preferences : IPreferences
+    {
+        private string _prefsFilePath;
+        private readonly IFileSystem _fileSystem;
+        private readonly IUserProfileDirectoryProvider _userProfileDirectoryProvider;
+
+        public string PreferencesFilePath
+        {
+            get
+            {
+                if (_prefsFilePath == null)
+                {
+                    string userProfileDirectory = _userProfileDirectoryProvider.GetUserProfileDirectory();
+                    _prefsFilePath = Path.Combine(userProfileDirectory, ".httpreplprefs");
+                }
+                return _prefsFilePath;
+            }
+        }         
+
+        public Preferences(IFileSystem fileSystem, IUserProfileDirectoryProvider userProfileDirectoryProvider)
+        {
+            _fileSystem = fileSystem;
+            _userProfileDirectoryProvider = userProfileDirectoryProvider;
+        }
+
+        public Dictionary<string, string> ReadPreferences(IReadOnlyDictionary<string, string> defaultPreferences)
+        {
+            var preferences = new Dictionary<string, string>(defaultPreferences);
+
+            if (_fileSystem.FileExists(PreferencesFilePath))
+            {
+                string[] prefsFile = _fileSystem.ReadAllLinesFromFile(PreferencesFilePath);
+
+                foreach (string line in prefsFile)
+                {
+                    int equalsIndex = line.IndexOf('=');
+
+                    if (equalsIndex < 0)
```

Comment:
is `=foo` a valid line?  equalsIndex would be 0, so I'd get an empty name?

Reply:
It wouldn't fail, but it wouldn't be useful. So I changed it to actively skip lines like that.